src/libxtp/threecenter_gwbse.cc

```diff
@@ -55,20 +52,32 @@ void TCMatrix_gwbse::MultiplyRightWithAuxMatrix(const Eigen::MatrixXd& matrix) {
   // Try to run the operation in a Nvidia GPU, otherwise is the default Openmp
   // implementation
 #if defined(USE_CUDA)
-  CudaPipeline cudaHandle;
-  try {
-    cudaHandle.right_matrix_tensor_mult(_matrix, matrix);
-  } catch (const std::runtime_error& error) {
-    XTP_LOG_SAVE(logDEBUG, _log)
-        << TimeStamp()
-        << " CUDA Multiplyrightwithauxmatrix failed due to: " << error.what()
-        << " Using default OpenMP implementation!" << flush;
-    this->MultiplyRightWithAuxMatrixOpenMP(matrix);
+  XTP_LOG_SAVE(logDEBUG, _log)
+      << TimeStamp()
+      << " Using CUDA/OpenMP for tensor matrix multiplication: " << flush;
+  // Preallocated memory for all the matrices
+  CudaPipeline cuda_pip;
+  const Eigen::MatrixXd& head = _matrix.front();
+  CudaMatrix cuma_A{head.rows(), head.cols()};
+  CudaMatrix cuma_B{matrix, cuda_pip.get_stream()};
+  CudaMatrix cuma_C{head.rows(), matrix.cols()};
+
+#pragma omp parallel for schedule(dynamic)
```

Comment:
Can you add a small comment explaining why only one GPU thread. If you have a link to somewhere even better